### PR TITLE
LdapMapStorageProvider to use a full inline class for MapStorage

### LINKED ISSUE #11373: New Storage: Make LdapMapStorageProvider use a full inline class form for the MapStorage implementation
### Description

Currently `LdapMapStorageProvider.getStorage` is using a lamba expression to return the `MapStorage` implementation. There is a request to use the full inline class form because `MapStorage` is intentionally not marked as @FunctionalInterface as there might be methods added in the future.

### PR DESCRIPTION
**Pull request by ahus1**
Closes #11373

### CODE REVIEW

**Review by vramik — APPROVED** [review]
Thanks @ahus1! The same is applies for `JpaMapStorageProvider`

File: model/map-ldap/src/main/java/org/keycloak/models/map/storage/ldap/LdapMapStorageProvider.java [hunk 1]
```diff
@@ -39,13 +40,18 @@ public void close() {
...
+        // MapStorage is not a functional interface, therefore don't try to convert it to a lambda as additional methods might be added in the future
+        //noinspection Convert2Lambda
+        return new MapStorage<V, M>() {
+            @Override
+            public MapKeycloakTransaction<V, M> createTransaction(KeycloakSession session) {
+                MapKeycloakTransaction<V, M> sessionTx = session.getAttribute(sessionTxPrefix + modelType.hashCode(), MapKeycloakTransaction.class);
+                if (sessionTx == null) {
+                    sessionTx = factory.createTransaction(session, modelType);
+                    session.setAttribute(sessionTxPrefix + modelType.hashCode(), sessionTx);
+                }
```
Comment (hmlnarik) [inline model/map-ldap/src/main/java/org/keycloak/models/map/storage/ldap/LdapMapStorageProvider.java:52]: Should this be under `synchronized` guard, like e.g. here? https://github.com/keycloak/keycloak/blob/b356618cc2e00ef3ae37f515c1d4b3bdbcde3f62/model/jpa/src/main/java/org/keycloak/connections/jpa/DefaultJpaConnectionProviderFactory.java#L160-L162

Comment (ahus1) [reply model/map-ldap/src/main/java/org/keycloak/models/map/storage/ldap/LdapMapStorageProvider.java:52]: There is no need to a guard: The returned MapStorage is stateless and lightweight. The only method being called is called with a KeycloakSession as parameter, and all calls within a KeycloakSession are non-concurrent.

Comment (ahus1) [reply model/map-ldap/src/main/java/org/keycloak/models/map/storage/ldap/LdapMapStorageProvider.java:52]: There is no need to a guard: The returned MapStorage is stateless and lightweight. The only method being called is called with a KeycloakSession as parameter, and all calls within a KeycloakSession are non-concurrent. 
@hmlnarik

Comment (hmlnarik) [reply model/map-ldap/src/main/java/org/keycloak/models/map/storage/ldap/LdapMapStorageProvider.java:52]: Same argument appiles here as in https://github.com/keycloak/keycloak/pull/11328/files#r856132221, resolving

**Review by hmlnarik — APPROVED** [review]

**Pull request merged**